core/services/workflows/workflow.go

```diff
@@ -9,15 +16,127 @@ type Capability struct {
 	Config map[string]any `yaml:"config"`
 }
 
-type Workflow struct {
+type workflowSpec struct {
 	Triggers  []Capability `yaml:"triggers"`
 	Actions   []Capability `yaml:"actions"`
 	Consensus []Capability `yaml:"consensus"`
 	Targets   []Capability `yaml:"targets"`
 }
 
-func Parse(yamlWorkflow string) (*Workflow, error) {
-	wf := &Workflow{}
-	err := yaml.Unmarshal([]byte(yamlWorkflow), wf)
+func (w *workflowSpec) steps() []Capability {
+	s := []Capability{}
+	s = append(s, w.Actions...)
+	s = append(s, w.Consensus...)
+	s = append(s, w.Targets...)
+	return s
+}
+
+type workflow struct {
+	*graph[*node]
+
+	triggers []*triggerCapability
+
+	spec *workflowSpec
+}
+
+type node struct {
+	Capability
+	dependencies     []string
+	cachedCapability capabilities.CallbackExecutable
+	cachedConfig     *values.Map
+}
+
+type triggerCapability struct {
+	Capability
+	cachedTrigger capabilities.TriggerCapability
```

Comment:
How do you feel about pre/postfixing with something like `instance` or `callable`, `pointer`, `delegate`?
If that sounds good, `triggerCapability` could also be named something like `triggerCapabilityClosure`, since its a callable struct that contains context that the wrapped function needs during execution.